subsys/bluetooth/services/alexa_gadgets/CMakeLists.txt

```diff
@@ -157,40 +157,52 @@ foreach(GROUP BASE NOTIFICATIONS ALERTS STATELISTENER SPEECHDATA MUSICDATA CUSTO
     endforeach()
 endforeach()
 
-zephyr_library_named(ALEXA_GADGETS_PROFILE)
+zephyr_library_named(alexa_gadgets_lib)
 
-zephyr_include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 zephyr_include_directories(${PROTO_COMPILATION_DIR})
 zephyr_include_directories(${NANOPB_PATH})
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_PROFILE_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS
+    ${CMAKE_CURRENT_SOURCE_DIR}/gadgets.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_PROFILE
     ${CMAKE_CURRENT_SOURCE_DIR}/gadgets_profile.c
     ${NANOPB_PATH}/pb_common.c
     ${NANOPB_PATH}/pb_encode.c
     ${NANOPB_PATH}/pb_decode.c
     ${BASE_PROTO_OUTPUTS}
 )
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_CAPABILITY_NOTIFICATIONS_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_CAPABILITY_NOTIFICATIONS
     ${NOTIFICATIONS_PROTO_OUTPUTS}
 )
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_CAPABILITY_ALERTS_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_CAPABILITY_ALERTS
     ${ALERTS_PROTO_OUTPUTS}
 )
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_CAPABILITY_STATELISTENER_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_CAPABILITY_STATELISTENER
     ${STATELISTENER_PROTO_OUTPUTS}
 )
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_CAPABILITY_SPEECHDATA_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_CAPABILITY_SPEECHDATA
     ${SPEECHDATA_PROTO_OUTPUTS}
 )
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_CAPABILITY_MUSICDATA_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_CAPABILITY_MUSICDATA
     ${MUSICDATA_PROTO_OUTPUTS}
 )
 
-zephyr_library_sources_ifdef(CONFIG_GADGETS_CAPABILITY_CUSTOM_ENABLE
+zephyr_library_sources_ifdef(CONFIG_BT_ALEXA_GADGETS_CAPABILITY_CUSTOM
     ${CUSTOM_PROTO_OUTPUTS}
 )
+
+#
+# Give the main application access to the generated proto headers
+#
+# NOTE: This library corresponds to APP_LINK_WITH_ALEXA_GADGETS_PROTO in Kconfig
+#       which is always set to "true".
```

Comment:
That is not true.

A user may decide to disable this by setting CONFIG_APP_LINK_WITH_ALEXA_GADGETS_PROTO=n`.

A use-case for doing so, is when a user defines a custom library (not the app lib) that requires linking with alexa_gadgets_proto lib.

In that case user can disable the app linking config, and instead do the linking directly in the CMakeLists.txt file, like:
```
target_link_libraries(my_custom_lib PRIVATE alexa_gadgets_proto)
```

Reply:
Copied from another cmakelists.txt without consideration. Changing wording to "default true" instead.